mass_mailing_partner/models/mail_mass_mailing_contact.py

```diff
@@ -75,7 +82,9 @@ def _set_name_email(self, vals):
         partner_id = vals.get('partner_id', self.partner_id.id)
         if not partner_id:
             return vals
-        partner = self.env['res.partner'].browse(partner_id)
-        vals['email'] = partner.email
-        vals['name'] = partner.name
+        partner = self.env['res.partner'].browse(partner_id, self._prefetch)
+        if vals.get("email", self.email) != partner.email:
```

Comment:
s/self.email/partner.email?

Reply:
The point here is to avoid updating the record if the email found on it is equal to the partner's, so AFAICS it's OK as it is: It gets the email passed through `vals`, and if none, it gets it from current record, to be able to compare it with the partner's.

Reply:
I'm talking for avoiding singleton error. Replace `self` (multi-record) by `partner` (single-record)

Reply:
This method [was not meant to be called on multi records before](https://github.com/OCA/social/pull/315/files#diff-6e2abeeb2e2cc273679a065d55fcd860R82), so that's not a problem here.

Reply:
OK, got it, although the variable names are confusing (specially `partner` meaning the other part).